cmake/HPX_SetupTarget.cmake

```diff
@@ -204,6 +203,11 @@ function(hpx_setup_target target)
     if(DEFINED HPX_LIBRARIES)
       set(hpx_libs ${hpx_libs} ${HPX_LIBRARIES})
     endif()
+
+    # add linker flags for libhpx_init.a for linux environment
+    if(CMAKE_SYSTEM_NAME STREQUAL Linux)
+      set_target_properties(${target} PROPERTIES LINK_FLAGS "-Wl,-wrap=__libc_start_main")
+    endif()
```

Comment:
This adds the overloaded entry point unconditionally. Is that what we want? Shouldn't this be done for executables only (or at least those modules that link against hpx_init)?

Reply:
Yes, this should be done for executables only. I will add another parameter to the if statement to add for target type executables.